examples/data-objects/diceroller/src/main.tsx

```diff
@@ -72,6 +72,14 @@ export class DiceRoller extends DataObject implements IDiceRoller, IFluidHTMLVie
 
     public get IFluidHTMLView() { return this; }
 
+    public static readonly factory = new DataObjectFactory<DiceRoller, undefined, undefined, IEvent>
```

Comment:
I don't think we should go this direction of hanging factories on `DataObject`s -- note #6399 which has a similar conversation about DDS (rather than `DataObject`) where the same arguments apply.

Perhaps even moreso here, since a `DiceRoller` has no need to know/refer to a factory which can produce it -- it has no `DiceRoller.create()` like the DDSs have.

Reply:
I agree that anything the class isn't using is considered extraneous and we should eliminate that. The only reason I wanted to include `DiceRoller` is to account for a scenario where a more complex `DataObject` is passed in as a `DynamicObjectType`. Perhaps this scenario isn't needed since it's purpose is the same thing as the previous test case testing DDSs? Or maybe there are other custom `DataObject`s that would be a better fit?

Reply:
@ChumpChief When we pass in the `DiceRoller` type for the `containerSchema`, the class needs to have its factory defined on it so that `FluidContainer` knows how to generate a new instance of it

Reply:
If we expect third parties to use DOs here, they'll either need to do this, or we'll need to lift that requirement in FluidContainer.  We can just drop the test for this PR, but we definitely need to have a discussion about reconciling this at some point.

Reply:
> @ChumpChief Matt Rakow FTE When we pass in the `DiceRoller` type for the `containerSchema`, the class needs to have its factory defined on it so that `FluidContainer` knows how to generate a new instance of it

I remember there was some reason that we didn't want the developer to pass the factory to use for the DO, rather than infer it based on convention? Was it just for simplicity (DOs are self-contained; they know how to create themselves, how to serialize themselves, etc.)?

Thinking through this... if we had to change the design in the future to be more configurable in a back-compat way, how would we do it? One potential: we add the factory as an optional parameter and when it's missing, we inspect the DO at runtime to see if it defines a factory, in which case it's called.

Reply:
I see -- I think let's go ahead with adding the factory as a member then.  I don't want to add options for factory vs. DO as I think this adds ambiguity without really improving the developer's experience.

Reply:
Thanks for all the feedback, I'll go ahead with adding factory member to `DiceRoller` then. We can come back and clarify the ambiguity more later on.